src/nix.rs

```diff
@@ -0,0 +1,338 @@
+
+        ParsedType::Value(n) => match n.to_value().unwrap() {
+            value::Value::Float(v) => Term::Num(v).into(),
+            value::Value::Integer(v) => Term::Num(v as f64).into(),
+            value::Value::String(v) => Term::Str(v).into(),
+            // TODO: How to manage Paths in nickel?
+            value::Value::Path(a, v) => Term::Str(v).into(),
+        },
+        ParsedType::Str(n) => {
+            let parts = n.parts();
+            //TODO: Do we actualy need the `Term::Str` variant in nickel AST?
+            Term::StrChunks(
+                parts
+                    .into_iter()
+                    .enumerate()
+                    .map(|(i, c)| match c {
+                        rnix::value::StrPart::Literal(s) => crate::term::StrChunk::Literal(s),
+                        rnix::value::StrPart::Ast(a) => crate::term::StrChunk::Expr(
+                            translate(a.first_child().unwrap(), file_id),
+                            i,
+                        ),
+                    })
+                    .collect(),
+            )
+            .into()
+        }
+        ParsedType::List(n) => Term::Array(
+            n.items().map(|elm| translate(elm, file_id)).collect(),
+            Default::default(),
+        )
+        .into(),
+        ParsedType::AttrSet(n) => {
+            use crate::parser::utils::{build_record, elaborate_field_path, FieldPathElem};
+            let fields: Vec<(_, _)> = n
+                .entries()
+                .map(|kv| {
+                    let val = translate(kv.value().unwrap(), file_id);
+                    let p: Vec<_> = kv
+                        .key()
+                        .unwrap()
+                        .path()
+                        .map(|p| match p.kind() {
+                            NODE_IDENT => FieldPathElem::Ident(
+                                rnix::types::Ident::cast(p).unwrap().as_str().into(),
+                            ),
+                            _ => FieldPathElem::Expr(translate(p, file_id)),
+                        })
+                        .collect();
+                    elaborate_field_path(p, val)
+                })
+                .collect();
+            build_record(fields, Default::default()).into()
+        }
+
+        // In nix it's allowed to define vars named `true`, `false` or `null`.
+        // Concidering it a bad feature and something probably never used, this parser don't manage
+        // it. Right now it doesn't throw error when defining these vars but only ignore the
+        // definition.
+        ParsedType::Ident(n) => match n.as_str() {
+            "true" => Term::Bool(true),
+            "false" => Term::Bool(false),
+            "null" => Term::Null,
+            id => Term::Var(id.into()),
+        }
+        .into(),
+        ParsedType::LegacyLet(_) => unimplemented!(), // Probably useless to suport it in a short term.
+        ParsedType::LetIn(n) => {
```

Comment:
What could be a nice addition: a comment explaining that because `let` are mutually recursive in Nix, you translate them as recursive records + pattern matching in Nickel. It might take some head scratching to understand it as it is :slightly_smiling_face: